test/language/expressions/object/literal-property-name-bigint.js

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2019 Igalia S.L, Toru Nagashima. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: >
+    BigInt in LiteralPropertyName must be valid and the property name must be
+    the string representation of the numeric value.
+esid: prod-PropertyName
+info: |
+    PropertyName[Yield, Await]:
+        LiteralPropertyName
+        ComputedPropertyName[?Yield, ?Await]
+
+    LiteralPropertyName:
+        IdentifierName
+        StringLiteral
+        NumericLiteral
+
+    NumericLiteral:
+        DecimalLiteral
+        DecimalBigIntegerLiteral
+
+    LiteralPropertyName: NumericLiteral
+        1. Let _nbr_ be the NumericValue of |NumericLiteral|.
+        1. Return ! ToString(_nbr_).
+features: [BigInt, class, destructuring-binding, let]
+---*/
+
+// Property
+
+let o = { 999999999999999999n: true }; // greater than max safe integer
+
+assert.sameValue(o["999999999999999999"], true,
+    "the property name must be the string representation of the numeric value.");
+
+o = { 1n: "foo" };
+assert.sameValue(o[1n], "foo");
+assert.sameValue(o[1], "foo");
+assert.sameValue(o["1"], "foo");
+
+// MethodDeclaration
+
+o = { 1n() { return "bar"; } };
+assert.sameValue(o[1n](), "bar");
```

Comment:
Also, following your pattern from previous code, I think it is enough to have only 1 assert with string key. I would also recommend add messages on `assert.sameValue`, as you've done. The message here could be `Method name must be the string representation of the numeric value`.